apps/studio/src/components/ResourceSelector/ResourceItem.tsx

```diff
@@ -1,86 +1,95 @@
 import type { IconType } from "react-icons"
-import { Suspense, useMemo } from "react"
-import { Box, HStack, Icon, Skeleton, Text } from "@chakra-ui/react"
-import { dataAttr } from "@chakra-ui/utils"
+import { Suspense } from "react"
+import { Icon, Text } from "@chakra-ui/react"
 import { Button } from "@opengovsg/design-system-react"
+import { ResourceType } from "@prisma/client"
 import { QueryErrorResetBoundary } from "@tanstack/react-query"
-import { ResourceType } from "~prisma/generated/generatedEnums"
 import { ErrorBoundary } from "react-error-boundary"
-import { BiData, BiFile, BiFolder, BiLink, BiLockAlt } from "react-icons/bi"
+import { BiData, BiFile, BiFolder, BiLink } from "react-icons/bi"
 
 import type { RouterOutput } from "~/utils/trpc"
 
 type ResourceItemProps = Pick<
-  RouterOutput["resource"]["getChildrenOf"]["items"][number],
-  "permalink" | "type"
+  RouterOutput["resource"]["getFolderChildrenOf"]["items"][number],
+  "permalink"
 > & {
-  isSelected: boolean
-  isDisabled: boolean
-  onResourceItemSelect: () => void
+  handleOnClick: () => void
+  type: ResourceType | undefined
+  isDisabled?: boolean
+  isHighlighted?: boolean
+}
+
+const getButtonProps = ({ isHighlighted }: { isHighlighted: boolean }) => {
+  if (isHighlighted) {
+    return {
+      color: "interaction.main.default",
+      bg: "interaction.muted.main.active",
+      _hover: {
+        color: "interaction.main.default",
+        bg: "interaction.muted.main.active",
+      },
+    }
+  }
+
+  return {
+    color: "base.content.default",
+  }
+}
+
+const getButtonIcon = ({
+  type,
+}: {
+  type: ResourceType | undefined
+}): IconType => {
+  switch (type) {
+    case ResourceType.CollectionLink:
+      return BiLink
+    case ResourceType.Folder:
+      return BiFolder
+    case ResourceType.CollectionPage:
+    case ResourceType.Page:
+      return BiFile
+    case ResourceType.Collection:
+      return BiData
+    default:
+      return BiData // Default to data icon
+  }
 }
 
 const SuspendableResourceItem = ({
   permalink,
-  type,
-  isSelected,
-  isDisabled,
-  onResourceItemSelect,
+  isHighlighted,
+  handleOnClick,
+  ...rest
 }: ResourceItemProps) => {
-  const icon: IconType = useMemo(() => {
-    switch (type) {
-      case ResourceType.CollectionLink:
-        return BiLink
-      case ResourceType.Folder:
-        return BiFolder
-      case ResourceType.CollectionPage:
-      case ResourceType.Page:
-        return BiFile
-      case ResourceType.Collection:
-        return BiData
-    }
-  }, [type])
+  const { type, ...restWithoutType } = rest
+
+  const buttonProps = getButtonProps({
+    isHighlighted: !!isHighlighted,
+  })
```

Comment:
this looks abit off to me - does the `active/focus` state not already account for the `highlighted` state? i thought the change would be to specify a base style and override on `_active`, for example

Reply:
OGPDS does but our design is slightly different so there's a number of overwriting here to fit it to our use case

Reply:
sounds good!